lib/logidze/meta.rb

```diff
@@ -3,18 +3,20 @@
 module Logidze # :nodoc:
   # Provide methods to attach meta information
   module Meta
-    def with_meta(meta, &block)
-      MetaTransaction.wrap_with(meta, &block)
+    def with_meta(meta, transactional: true, &block)
```

Comment:
It's definitely up to you, but we could reduce the repetitive part in a following way:

```ruby
klass = transactional ? MetaWithTransaction : MetaWithoutTransaction
klass.wrap_with(meta, &block)
```